### PR TITLE
[search] Fixing the dimension mismatch error and bool comparison issue

### PR DESCRIPTION
**Pull request by neccam**
This commit is fixing two bugs I had while running the code with PyTorch 1.3.0
  - When the alpha value is set to -1 or lower (i.e. alpha = -1) I was having dimension mismatch error in the line 281 (log_probs += topk_log_probs.view(-1).unsqueeze(1)). Adding the lines 298-299 fixed that error.
  - The comparisons of bool tensors with int values, i.e. end_condition.eq(0), was not working properly (which might have something to do with the new pytorch version). As a result, the beam search was predicting sequences without \<eos\> tokens and lots of \<pad\>s. I replaced the int values with their boolean counterparts the lines 318, 320 and 348 to fix this issue.

I also did some housekeeping changes
  - Changed the creation/initialisation of topk_log_probs tensor to match the shape of topk_scores.
  - Combined the results dict creation into a single command at lines 238-242.
  - I added a tensor clone() call in the line 282 for keeping the code safe for future changes. (Changes in curr_scores were effecting the log_probs tensor)
  - I added an if statement to check how many \<eos\> tokens are there in a predicted sequence.  If it has more than one \<eos\>, it means that the prediction should have already been added to the hypotheses, so you don't have to add them again.

### CODE REVIEW

**Comment by neccam** [comment]
The build error is:
```
error: six 1.13.0 is installed but six==1.12 is required by {'astroid'}
The command "python setup.py install" failed and exited with 1 during .
```
which, I believe, has nothing to do with my changes.

**Comment by juliakreutzer** [comment]
Thanks so much! Amazing job, this is really helpful :sunglasses: 

The six error seems to be caused by this issue in the astroid package (for pylint): https://github.com/PyCQA/astroid/issues/718 - it looks like they're working on it and it should be fixed soon. I'll rerun and merge the changes as soon as that issue is resolved.

File: joeynmt/search.py [hunk 1]
```diff
@@ -328,10 +329,13 @@ def beam_search(
                 finished_hyp = is_finished[i].nonzero().view(-1)
                 # store finished hypotheses for this batch
                 for j in finished_hyp:
-                    hypotheses[b].append((
-                        topk_scores[i, j],
-                        predictions[i, j, 1:])  # ignore start_token
-                    )
+                    # Check if the prediction has more than one EOS.
+                    # If it has more than one EOS, it means that the prediction should have already
+                    # been added to the hypotheses, so you don't have to add them again.
```
Comment (juliakreutzer) [inline joeynmt/search.py]: Thanks for adding comments about this as well, this is will help others to understand what's going on :+1:

**Comment by neccam** [comment]
Thanks for your nice comments :) Would you be able to give me permission to merge?

**Comment by juliakreutzer** [comment]
Merged it through an extra branch with edits to six requirements (temporary fix): #74 
Again, thanks so much for your contribution :clap:

**Pull request merged**